test/integration/action_execution.go

```diff
@@ -255,6 +255,48 @@ var _ = Describe("Action Executor ", func() {
 		})
 	})
 
+	// Horizontal scale test
+	Describe("Executing horizontal scale action on a deployment", func() {
+		It("should match the expected replica number", func() {
+			// create a deployment with 2 replicas
+			dep, err := createDeployResource(kubeClient, depSingleContainerWithResources(namespace, "", 2, false, false, false))
+			framework.ExpectNoError(err, "Error creating test resources")
+
+			pod, err := getDeploymentsPod(kubeClient, dep.Name, namespace, "")
+			framework.ExpectNoError(err, "Error getting deployments pod")
+			// This should not happen. We should ideally get a pod.
+			if pod == nil {
+				framework.Failf("Failed to find a pod for deployment: %s", dep.Name)
+			}
+
+			targetNodeName := getTargetSENodeName(f, pod)
+			if targetNodeName == "" {
+				framework.Failf("Failed to find a pod for deployment: %s", dep.Name)
+			}
+
+			// Test the provision action
+			_, err = actionHandler.ExecuteAction(newActionExecutionDTO(proto.ActionItemDTO_PROVISION,
+				newTargetSEFromPod(pod), nil), nil, &mockProgressTrack{})
+			framework.ExpectNoError(err, "Failed to execute provision action")
+
+			newDep, err := waitForDeployment(kubeClient, dep.Name, dep.Namespace)
```

Comment:
`waitForDeployment()` is time wasting check slowing down the test. It will wait for the pod to become ready.
You can simply wait on check for the deployment ->spec.replicas being updated to desired value.

Reply:
Sure, I'm going to add a new function `waitForDeploymentToUpdateReplica` to wait to check the deployment replica, which will be very similar to `waitForDeployment`, but as they're checking the different fields of the deployment, it still makes sense for me.